tests/TestVersion.py

```diff
@@ -3,30 +3,53 @@
 
 from UM.Version import Version
 from unittest import TestCase
+import pytest
 
+major_versions = [Version("1"), Version(b"1"), Version(1), Version([1]), Version(["1"]), Version("1."), Version("MOD-1"), Version("1B")]
+major_minor_versions = [Version("1.2"), Version("1-2"), Version("1_2"), Version(b"1.2"), Version("1.2BETA"), Version([1, 2]), Version(["1", "2"]), Version([1, "2"]),]
+major_minor_revision_versions = [Version("1.2.3"), Version("1.2.3BETA"), Version("1_2-3"), Version(b"1.2.3"), Version([1, 2, 3]), Version(["1", "2", "3"]), Version([1, "2", 3]), Version("MOD-1.2.3")]
 
-class TestVersion(TestCase):
-    def test_version(self):
-        self.assertTrue(Version("1.1.0") > Version("1.0.0"))
-        self.assertTrue(Version("2.1.0") > Version("1.5.0"))
-        self.assertTrue(Version("2.1.0") > Version("1.1.0"))
-        self.assertTrue(Version("1.1.2") > Version("1.1.1"))
-        self.assertTrue(Version("1.1.2") == Version("1.1.2"))
-
-    def test_version2(self):
-        self.assertTrue(Version("1.1.0-alpha.2") > Version("1.1.0-alpha.1"))
-        self.assertTrue(Version("1.1.0-alpha.2") == Version("1.1.0-alpha.2"))
-        self.assertFalse(Version("1.1.0-alpha.2") > Version("1.1.1-alpha.1"))
-        self.assertTrue(Version("2.1.0-beta") > Version("1.5.0"))
-        self.assertFalse(Version("1.1.2-beta.2") > Version("1.1.2.alpha.1"))
-        self.assertTrue(Version("1.0.0") > Version("1.0.0-alpha.7"))
-
-    def test_malformed_versions(self):
-        v1 = Version("This is a strange version number")
-        v2 = Version("2.3.4.5-beta.2.3")
-        v1 > v2  # Just don't crash
-
-    def test_other_comparisons(self):
-        self.assertFalse(Version("1.1.0-beta.2") == Version("1.1.0-alpha.2"))
-        self.assertFalse(Version("1.1.0-beta.2") > Version("1.1.0-alpha.2"))
-        self.assertFalse(Version("1.1.0-beta.2") < Version("1.1.0-alpha.2"))
+
+def check_version_equals(first_version: Version, second_version: Version):
+    assert first_version == second_version
+    assert first_version.getMajor() == second_version.getMajor()
+    assert first_version.getMinor() == second_version.getMinor()
+    assert first_version.getRevision() == second_version.getRevision()
+
+
+@pytest.mark.parametrize("first_version", major_versions)
+@pytest.mark.parametrize("second_version", major_versions)
+def test_major_equals(first_version, second_version):
+    check_version_equals(first_version, second_version)
+
+
+@pytest.mark.parametrize("first_version", major_minor_versions)
+@pytest.mark.parametrize("second_version", major_minor_versions)
+def test_major_and_minor_equals(first_version, second_version):
+    check_version_equals(first_version, second_version)
```

Comment:
Something goes wrong with these tests: I added print("Compare: "+ str(first_version) + " _ " + str(second_version))
All the time it compares same versions, I get each time 
Compare: 1.2.0 _ 1.2.0
Compare: 1.2.0 _ 1.2.0
...

Reply:
This is intentional; it's creating a bunch of Version objects in all the ways that it can and subsequently compare them. What this test checks is that Version("1.1") is indeed the same as Version([1,1])